### PR TITLE
Selection-playlist-test

### PR DESCRIPTION
**Pull request by RenBabiuch**

### CODE REVIEW

File: src/test/java/Page_Objects/GamePage.java [hunk 1]
```diff
@@ -12,21 +12,28 @@
...
+    @FindBy(css = "[data-test='button-exit-song']")
+    private WebElement exitButton;
+
+
     public GamePage(WebDriver driver) {
         this.driver = driver;
         PageFactory.initElements(driver, this);
         wait = new WebDriverWait(driver, Duration.ofSeconds(10));
         actions = new Actions(driver);
+        timer = new Timer();
```
Comment (Asvarox) [inline src/test/java/Page_Objects/GamePage.java]: is this needed

File: src/test/java/Page_Objects/SongListPage.java [hunk 2]
```diff
@@ -114,7 +116,7 @@ public void goToPlaylist(String playlistName) {
     public void expectPlaylistToBeSelected(String playlistName) {
         String playlistSelectedAttribute = playlistElement(playlistName).getAttribute("data-selected");
 
-        Assertions.assertEquals(playlistSelectedAttribute, "true");
+        Assertions.assertEquals("true", playlistSelectedAttribute, "Expected playlist is not selected");
```
Comment (Asvarox) [inline src/test/java/Page_Objects/SongListPage.java]: it's always a good idea to add what exactly was expected and what is the actual result. E.g. this error message could be
```
"Expected playlist " + playListName + " to be selected but it is not. data-selected=" + playlistSelectedAttribute + " instead"
```

File: src/test/java/Page_Objects/SongListPage.java [hunk 3]
```diff
@@ -124,4 +126,30 @@ public void approveSongByKeyboard() {
     public boolean isSongMarkedWithDuetIcon(String songID) {
         return getSongElement(songID).findElement(By.cssSelector("[data-test='multitrack-indicator']")).isDisplayed();
     }
+
+    public void expectSongToBeMarkedAsPlayedToday(String songID) {
+        String playedStatusText = getSongElement(songID).findElement(By.cssSelector("[data-test='song-stat-indicator']")).getText();
+        String playedStatusSubs = playedStatusText.substring(0, 1).toUpperCase() + playedStatusText.substring(1).toLowerCase();
```
Comment (Asvarox) [inline src/test/java/Page_Objects/SongListPage.java]: why not just put everything toLowerCase, would be simpler

File: src/test/java/Page_Objects/SongListPage.java [hunk 4]
```diff
@@ -124,4 +126,30 @@ public void approveSongByKeyboard() {
     public boolean isSongMarkedWithDuetIcon(String songID) {
         return getSongElement(songID).findElement(By.cssSelector("[data-test='multitrack-indicator']")).isDisplayed();
     }
+
+    public void expectSongToBeMarkedAsPlayedToday(String songID) {
+        String playedStatusText = getSongElement(songID).findElement(By.cssSelector("[data-test='song-stat-indicator']")).getText();
+        String playedStatusSubs = playedStatusText.substring(0, 1).toUpperCase() + playedStatusText.substring(1).toLowerCase();
+
+        Assertions.assertEquals("Played today", playedStatusSubs, "Played today status should be visible for the song");
```
Comment (Asvarox) [inline src/test/java/Page_Objects/SongListPage.java]: similar to the above comment about the error message, you could include the song id and what is the actual text here

File: src/test/java/tests/SelectionPlaylistTest.java [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.openqa.selenium.chrome.ChromeDriver;
+
+public class SelectionPlaylistTest {
+
+    private ChromeDriver driver;
+    private TestBase testBase;
+    String compSongID = "xavier-rudd-stoney-creek";
+    String almSongID = "electric-light-orchestra-bluebird";
```
Comment (Asvarox) [inline src/test/java/tests/SelectionPlaylistTest.java]: I think it's better to just use single song that is not on selection playlist - smaller chance that any of the tests fail

File: src/test/java/tests/SelectionPlaylistTest.java [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+        // Step 3: Pick up and play random popular song from Selection playlist
+        testBase.getSongListPage().goToPlaylist(selectionPlaylist);
+        testBase.getSongListPage().doesPlaylistContainSongsMarkedAsPopular();
+
+        String popSong = testBase.getSongListPage().getSelectedSongID();
+
+        testBase.getSongListPage().approveSongByKeyboard();
+        testBase.getSongPreviewPage().goNext();
+        testBase.getSongPreviewPage().goToPlayTheSong();
+        Assertions.assertTrue(testBase.getGamePage().lyricsContainerElement(0).isDisplayed(), "The song lyrics for player should be visible");
```
Comment (Asvarox) [inline src/test/java/tests/SelectionPlaylistTest.java]: This check doesn't seem to be necessary for this test

File: src/test/java/Page_Objects/GamePage.java (hunk 1 above)
Comment (RenBabiuch) [reply src/test/java/Page_Objects/GamePage.java]: No, you're right - I forgot to delete it. Thanks :)

File: src/test/java/Page_Objects/SongListPage.java (hunk 2 above)
Comment (RenBabiuch) [reply src/test/java/Page_Objects/SongListPage.java]: Ok, i got it. Thanks for tip :)

File: src/test/java/Page_Objects/SongListPage.java (hunk 3 above)
Comment (RenBabiuch) [reply src/test/java/Page_Objects/SongListPage.java]: good point, thanks :)

**Review by Asvarox — APPROVED** [review]

**Pull request merged**